src/EditorFeatures/CSharpTest/ExtractClass/ExtractClassTests.cs

```diff
@@ -579,6 +579,51 @@ int Method()
             await TestExtractClassAsync(input, expected);
         }
 
+        [Fact]
+        public async Task TestFileHeader2()
+        {
+            var input = @"
+<Workspace>
+    <Project Language=""C#"">
+        <Document FilePath=""Test.cs"">// this is my document header
+// that should be ignored
+
+class Test
+{
+    int [||]Method()
+    {
+        return 1 + 1;
+    }
+}
+        </Document>
+    </Project>
+</Workspace>";
+
+            var expected = @"
+<Workspace>
+    <Project Language=""C#"">
+        <Document FilePath=""Test.cs"">// this is my document header
+// that should be ignored
```

Comment:
nit

```suggestion
// that should be changed
```

Reply:
um.. I don't understand?

This header is ignored because its overridden by the real document header from the setting, when creating the new file, but the existing file is not affected at all. The "File banner incorrect" analyzer would presumably trigger on this, however.